### ISSUE TITLE
Can't build for macOS, Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified

### ISSUE THREAD

**Issue by Piinks** [flutter/flutter#71613]
hot_mode_dev_cycle_macos_target__benchmark is broken currently and complaining of a cert issue

https://flutter-dashboard.appspot.com/#/build

Added verbose logging in https://github.com/flutter/flutter/pull/71610 to track down the issues.

```
Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified.
```

**Comment by jmagman** [comment]
This isn't very helpful, but it passed on my machine:
```
$ cd dev/integration_tests/flutter_gallery/
$ flutter build macos
$ cd macos
$ xcrun xcodebuild -workspace Runner.xcworkspace -configuration Debug -scheme Runner VERBOSE_SCRIPT_LOGGING=YES COMPILER_INDEX_STORE_ENABLE=NO EXCLUDED_ARCHS=arm64 PROVISIONING_PROFILE_SPECIFIER="match Development *" 
```
```
[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Users/m/Applications/Xcode-12-3-beta1.app/Contents/Developer
    • Xcode 12.3, Build version 12C5020f
    • CocoaPods version 1.10.0
[✓] Connected device (3 available)
    • macOS (desktop)  • macos      • darwin-x64     • Mac OS X 10.15.7 19H15 darwin-x64
```

**Comment by jmagman** [comment]
We've seen the failure on mac-1, mac-3, and mac-14.

On mac-1 it reproduces:
```
$ cd dev/integration_tests/flutter_gallery
$ ../../../bin/flutter build macos
```

**Comment by yusufm** [comment]
Is this just a config issue, rather than cert problem?

/Users/flutter/.cocoon/flutter/dev/integration_tests/flutter_gallery$ ../../../bin/flutter build macos
note: Using new build system                                            
note: Planning build                                                    
note: Constructing build description                                    
error: Pods-Runner does not support provisioning profiles. Pods-Runner does not support provisioning profiles, but provisioning profile match Development * has been manually specified. Set the provisioning profile value to "Automatic" in the build settings editor. (in target 'Pods-Runner' from project 'Pods')
** BUILD FAILED **                                                      
                                                                        
Building macOS application...                                           
Build process failed

**Comment by jmagman** [comment]
@yusufm I don't think iOS builds have failed on these machines, just the macOS build.  So some kind of environment issue, not sure what though.

**Comment by jmagman** [comment]
I don't really know how these machines are set up or where that provisioning profile env variable is set up.

**Comment by yusufm** [comment]
$ xcodebuild -version
Xcode 11.3
Build version 11C29

**Comment by jmagman** [comment]
@bkonyi is seeing this on his machine after upgrading CocoaPods.

**Comment by jmagman** [comment]
Still trying to figure out how this is happening on machines other than the devicelab.  It seems to be happening there because those machines have environment variables set to force manual code signing.
```
FLUTTER_XCODE_PROVISIONING_PROFILE_SPECIFIER=match Development *
FLUTTER_XCODE_DEVELOPMENT_TEAM=<id>
FLUTTER_XCODE_CODE_SIGN_STYLE=Manual
```
It reproduces from the command line (which uses those build settings), but not Xcode (which doesn't and is auto signed).

But I wouldn't think @bkonyi would be forcing that on...

**Comment by jmagman** [comment]
@bkonyi has the exact same environment variables set!  My guess at the moment is that there's still something wrong with the cert, maybe it's just set up for iOS development and not macOS development?

**Comment by jmagman** [comment]
It builds when I `unset FLUTTER_XCODE_PROVISIONING_PROFILE_SPECIFIER`.  There's something wrong with `match Development *` on our infra setup.
\cc @xster Do you have any insight on this?  Is `match Development *` some fastlane thing?

**Comment by yusufm** [comment]
Yes, the cert and profiles were generated via fastlane match

**Comment by jmagman** [comment]
Yeah the cert is there, there's something wrong with `match Development *`?  When I totally remove `PROVISIONING_PROFILE_SPECIFIER` by removing `FLUTTER_XCODE_PROVISIONING_PROFILE_SPECIFIER ` it seems to sign.

**Comment by jmagman** [comment]
@xster reported this in https://github.com/flutter/flutter/issues/71232 trying to migrate devicelab to LUCI, that should probably be forward duped to this if we think it's the same, this has more info.

That failure was on CoocaPods 1.9.3, and it was before the certs exired.

**Comment by jmagman** [comment]
I worked around this with https://github.com/flutter/flutter/pull/71860.  It seems specific to our CI setup.  Let's close this and reinvestigate if we get any user reports.

**Comment by jmagman** [comment]
I'm going to reopen this, https://github.com/flutter/flutter/pull/71860 was such a lame "workaround" (removing a feature users actually want https://github.com/flutter/flutter/issues/81377).

**Comment by jmagman** [comment]
To test locally:
```
$ cd dev/devicelab
$ FLUTTER_XCODE_PROVISIONING_PROFILE_SPECIFIER="match Development *" dart bin/run.dart -t hot_mode_dev_cycle_macos_target__benchmark
```
